gulp/createskin.js

```diff
@@ -0,0 +1,206 @@
+'use strict';
+
+var lazyReq = require('lazy-req')(require);
+var inquirer = require('inquirer');
+var fs = require('fs');
+var path = require('path');
+var through = lazyReq('through2');
+
+module.exports = function (gulp, gutil) {
+
+    /**
+     * This function should match all validations for studio skin creation. As of 16.4, studio skin creation does not
+     * allow underscores,spaces or dashes in skin name but allows them on the title.
+     */
+    function validate(value, regex, name, len) {
+        if (value === '') {
+            return 'Please provide a valid value';
+        }
+        var cases = value.match(regex);
+        if (cases === null) {
+            return 'Invalid value, please try again.';
+        }
+        if (value.length > len) {
+            return name + ' too long. Please use '+len+' chars or less';
+        }
+        return true;
+    }
+
+    /**
+     * Exports core plugin skins for sdk if core plugins are not already downloaded.
+     */
+    function downloadCorePlugins() {
+
+        var putils = require('../lib/plugin-utils');
+        var server = require('../lib/plugin-server.js')(gulp, gutil).getServer();
+        if (fs.existsSync(server.coreOutputDir())) {
+            if (gutil.env['debug']) {
+                putils.logDebug(gutil, "Core plugins exist in directory " + server.coreOutputDir() + "/ . Skipping download...");
+            }
+            //Start skin creation process
+            return createNewSkin();
+        }
+        var stream = through().obj();
+        var pluginExport = require('../lib/plugin-export.js')(gulp, gutil);
+        //Download core plugins
+        pluginExport.exportPlugin(server, {
+            pluginType: 'core',
+            doClear: false,
+            verboseMode: gutil.env['verbose'],
+            debugMode: gutil.env['debug'],
+            coreOutputDir: server.coreOutputDir()
+        }, undefined, function(err) {
+            if (err) {
+                putils.logError(gutil, "Error downloading core plugins! Error: "+err.message);
+                process.exit(1);
+            }
+            //Start skin creation process
+            return createNewSkin();
+        }).pipe(stream);
+    }
+
+    /**
+     * Interactive create skin function. Allows to create a brand new skin with responsive and non responsive support. 
+     * SDK skins are also valid skin parents. Creates a blank skin directory structure under skin base dir of sdk project.
+     */
+    function createNewSkin() {
+        var skinLib = require('../lib/skin')(gulp, gutil);
+        var putils = require('../lib/plugin-utils');
```

Comment:
Isn't it better to have this require statement globally so that you don't have to declare it again.